google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/testing/FakeStorageRpc.java

```diff
@@ -151,7 +151,9 @@ public StorageObject create(StorageObject object, InputStream content, Map<Optio
         continue;
       }
       so.setSize(size(so));
-      values.add(so);
+      if (so.getBucket().equals(bucket)) {
```

Comment:
Saw this after, but it can help to modify the first conditional, to short circuit sooner in the sequence of code.

```
if (!so.getBucket().equals(bucket) || !so.getName().startsWith(prefix)) {
        continue;
}
```

Reply:
Ohhh i see will addressed it by creating another PR